windows/src/engine/kmcomapi/util/utilkeymanoption.pas

```diff
@@ -323,7 +323,7 @@ procedure TUtilKeymanOptionEntry.CustomSave;
       if OpenKey('\'+SRegKey_WindowsRun_CU, True) then
       begin
         if FBoolValue then
-          WriteString(SRegValue_WindowsRun_Keyman, '"'+TKeymanPaths.KeymanDesktopInstallPath(TKeymanPaths.S_KMShell)+'" -s')
+          WriteString(SRegValue_WindowsRun_Keyman, '"'+TKeymanPaths.KeymanDesktopInstallPath(TKeymanPaths.S_KMShell)+'" -boot')
```

Comment:
Do we need to reapply the `-boot` flag to the setting when we upgrade from an earlier version of Keyman? AFAICT, the registry setting is not rewritten during the upgrade.

Reply:
The way I read it is that `RunTools.PrepareForReboot` Deletes the key `SRegValue_WindowsRun_Keyman`.

[// Delete the Keyman autostart; it will be reinstated when setup finishes](https://github.com/keymanapp/keyman/blob/27bf5355c50e1e027c80b79f62bf64c1cf7c09c5/windows/src/desktop/setup/RunTools.pas#L770-L784)


Then it is setup again with `firstrun` via the options being passed through the command line ..... and the those options being applied. I didn't step all the way through. 

I will add a test to this attaching an old build with the -s.